Sources/FireKit/UserSessionManager.swift

```diff
@@ -9,224 +9,232 @@ import Foundation
-    print("PTUserSessionManager: Auth State Changed")
-    if let user = user {
-      print("PTUserSessionManager: Auth State Changed -- User")
-      // User found -- call completion to handle any post-reauth workflow
-      onAuthSessionStart(user: user, error: nil)
-    } else if defaultToAnonymousUser {
-      print("PTUserSessionManager: Auth State Changed -- No User (Anon)")
-      // No user logged in and we want to default to anonymous user. Start an Anonymous User Session
-      startAnonymousSession()
-    } else if state == .authenticated {
-      print("PTUserSessionManager: Auth State Changed -- No User (Signout)")
-      // A user just signed out (i.e. we've gone from user to no user)
-      state = .unauthenticated
-      userID = nil
-      onAuthSessionEnd()
-    } else {
-      print("PTUserSessionManager: Auth State Changed -- No User")
-      state = .unauthenticated
-      userID = nil
-    }
-  }
-  
-  open func startAuthListener() {
-    print("PTUserSessionManager: Auth Listener Started")
-    Auth.auth().addStateDidChangeListener(onAuthStateChanged)
-  }
-  
-  // MARK: - Password Changes
-  
-  open func changePasswordRequest(replace oldPassword: String, with newPassword: String, completion: ((Result<Void, Error>) -> Void)? = nil) {
-    let user = Auth.auth().currentUser
-    let credentials = EmailAuthProvider.credential(withEmail: email, password: oldPassword)
-    user?.reauthenticate(with: credentials, completion: { (result, error) in
-      if let error = error {
-        completion?(.failure(error))
-      } else {
-        // User re-authenticated.
-        user?.updatePassword(to: newPassword){ error in
-          if let error = error { completion?(.failure(error)) }
-          else { completion?(.success(())) }
+    
+    @Published public var userReady: Bool = false
+    @Published public var isNew = false
+    @Published public var state: PTAuthenticationState = .initializing
+    
+    @Published public var userID: String? {
+        didSet { userReady = getUserReady() }
+    }
+    
+    let defaultToAnonymousUser: Bool
+    let onAuth: ((User) -> Void)?
+    
+    public init(defaultToAnonymousUser: Bool = false, onAuth: ((User) -> Void)? = nil ){
+        self.defaultToAnonymousUser = defaultToAnonymousUser
+        self.onAuth = onAuth
+    }
+    
+    public var initialized: Bool {
+        state != .initializing
+    }
+    
+    public var isAnonymous: Bool {
+        Auth.auth().currentUser?.isAnonymous ?? false
+    }
+    
+    public var email: String {
+        Auth.auth().currentUser?.email ?? ""
+    }
+    
+    
+    // MARK: - Exposed Callbacks
+    
+    open func onAnonymousSessionStart() { }
+    open func getUserReady() -> Bool {
+        userID != nil
+    }
+    
+    // Called on new FirebaseAuth sessions, triggered by Auth.stateChangedListener started in startAuthListener
+    open func onAuthSessionStart(user: FirebaseAuth.User?, error: Error?) {
```

Comment:
user? ve error? olmasına gerek yok, sessionStart'sa zaten user var. not optional olsun, error kalksın

Reply:
fair